syft/pkg/package.go

```diff
@@ -17,17 +17,18 @@ import (
 // Package represents an application or library that has been bundled into a distributable format.
 // TODO: if we ignore FoundBy for ID generation should we merge the field to show it was found in two places?
 type Package struct {
-	id        artifact.ID      `hash:"ignore"`
-	Name      string           // the package name
-	Version   string           // the version of the package
-	FoundBy   string           `hash:"ignore" cyclonedx:"foundBy"` // the specific cataloger that discovered this package
-	Locations file.LocationSet // the locations that lead to the discovery of this package (note: this is not necessarily the locations that make up this package)
-	Licenses  LicenseSet       // licenses discovered with the package metadata
-	Language  Language         `hash:"ignore" cyclonedx:"language"` // the language ecosystem this package belongs to (e.g. JavaScript, Python, etc)
-	Type      Type             `cyclonedx:"type"`                   // the package type (e.g. Npm, Yarn, Python, Rpm, Deb, etc)
-	CPEs      []cpe.CPE        `hash:"ignore"`                      // all possible Common Platform Enumerators (note: this is NOT included in the definition of the ID since all fields on a CPE are derived from other fields)
-	PURL      string           `hash:"ignore"`                      // the Package URL (see https://github.com/package-url/purl-spec)
-	Metadata  interface{}      // additional data found while parsing the package source
+	id           artifact.ID            `hash:"ignore"`
+	Name         string                 // the package name
+	Version      string                 // the version of the package
+	FoundBy      string                 `hash:"ignore" cyclonedx:"foundBy"` // the specific cataloger that discovered this package
+	Locations    file.LocationSet       // the locations that lead to the discovery of this package (note: this is not necessarily the locations that make up this package)
+	Licenses     LicenseSet             // licenses discovered with the package metadata
+	Language     Language               `hash:"ignore" cyclonedx:"language"` // the language ecosystem this package belongs to (e.g. JavaScript, Python, etc)
+	Type         Type                   `cyclonedx:"type"`                   // the package type (e.g. Npm, Yarn, Python, Rpm, Deb, etc)
+	CPEs         []cpe.CPE              `hash:"ignore"`                      // all possible Common Platform Enumerators (note: this is NOT included in the definition of the ID since all fields on a CPE are derived from other fields)
+	PURL         string                 `hash:"ignore"`                      // the Package URL (see https://github.com/package-url/purl-spec)
+	Dependencies DependencyCompleteness `hash:"ignore"`                      // the completeness of the dependency information for this package
```

Comment:
I think I said this above, but I don't think `Dependencies` is the right name for this field. `Licenses` is of type `LicenseSet`, which is a set of licenses. `CPEs` is a slice of `cpe.CPE`. `Locations` is a `LocationSet`. `Dependencies` is an enum about how complete the dependencies are. It seems to break an established pattern.